pkg/tun/tun_linux.go

```diff
@@ -0,0 +1,153 @@
+// tun_linux.go: Open a Tunnel (L3 virtual interface) using the Universal TUN/TAP device driver.
+package tun
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/datawire/dlib/dexec"
+	"github.com/telepresenceio/telepresence/v2/pkg/tun/buffer"
+)
+
+const devicePath = "/dev/net/tun"
+
+type Device struct {
+	*os.File
+	name  string
+	index uint32
+}
+
+// OpenTun creates a new TUN device and ensures that it is up and running.
+func OpenTun() (*Device, error) {
+	// https://www.kernel.org/doc/Documentation/networking/tuntap.txt
```

Comment:
That link is outdated; the kernel docs got restructured a couple of years ago.  The current link is https://www.kernel.org/doc/html/latest/networking/tuntap.html

Reply:
or https://www.kernel.org/doc/html/latest/_sources/networking/tuntap.rst.txt if you like plaintext